CMakeLists.txt

```diff
@@ -240,24 +240,26 @@ if(USE_MKLDNN)
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /EHsc /Gy /MT")
   endif()
 
-  set(MKLDNN_BUILD_TESTS OFF CACHE INTERNAL "" FORCE)
-  set(MKLDNN_BUILD_EXAMPLES OFF CACHE INTERNAL "" FORCE)
-  set(MKLDNN_ARCH_OPT_FLAGS "" CACHE INTERNAL "" FORCE)
-  set(MKLDNN_ENABLE_JIT_PROFILING OFF CACHE INTERNAL "" FORCE)
-  set(MKLDNN_LIBRARY_TYPE STATIC CACHE INTERNAL "" FORCE)
-  set(DNNL_ENABLE_CONCURRENT_EXEC ON CACHE INTERNAL "" FORCE)
-
-  if(NOT USE_OPENMP)
+  function(load_mkldnn)
```

Comment:
Why need put them into a function?

Reply:
Functions have their own scope in cmake. We don't want to change any of the settings outside of this function (such as the install target name for the include files)

Reply:
So it's really about setting the `CMAKE_INSTALL_INCLUDEDIR` differently for mkldnn which is introduced in this PR